sycl/doc/extensions/experimental/sycl_ext_oneapi_non_uniform_groups.asciidoc

```diff
@@ -423,225 +424,102 @@ if (will_branch)
-if (partition.leader()){
-  buf[partition.get_group_id()] = result;
 }
 ----
 
-A `fixed_size_group` can be used to provide an interface accepting a specific
-number of work-items:
 
-[source, c++]
-----
-void func_that_needs_4_work_items(sycl::ext::oneapi::experimental::fixed_size_group<4> group);
-----
+=== Chunk
 
-
-=== Tangle-Groups
-
-A tangle-group is a non-contiguous subset of a group representing work-items
-executing in a tangle. A tangle-group can therefore be used to capture all
-work-items currently executing the same control flow.
+A `chunk` represents a contiguous collection of work-items created by
+subdividing a group into equally sized parts, such that each work-item is a
+member of exactly one partition.
+The size of a chunk is a static (compile-time) property.
 
 
 ==== Creation
 
-New tangle-groups are created by partitioning an existing group, using the
-`get_tangle_group` free-function.
-
-NOTE: Creating a tangle-group may implicitly synchronize members of the
-`tangle_group` on some devices, since it may be necessary to wait for
-work-items to reconverge. For consistency, this synchronization is required by
-all implementations.
+A new `chunk` can only be created by partitioning an existing group with the
+`chunked_partition` function.
 
 [source, c++]
 ----
 namespace ext::oneapi::experimental {
 
-template <typename Group>
-tangle_group get_tangle_group(Group group);
+template <size_t ChunkSize, typename ParentGroup>
+chunk<ChunkSize, Group> chunked_partition(ParentGroup parent);
 
 } // namespace ext::oneapi::experimental
 ----
 
-_Constraints_: Available only if `sycl::is_group_v<std::decay_t<Group>> &&
-std::is_same_v<Group, sycl::sub_group>` is true.
+_Constraints_: Available only if `sycl::is_group_v<std::decay_t<ParentGroup>>`
+is true and `ParentGroup` is a `sycl::sub_group` or `sycl::chunk`.
+`ChunkSize` must be positive and a power of 2.
+
+_Preconditions_: `ChunkSize` must be less than or equal to the result of
+`parent.get_max_local_range()`.
+`parent.get_local_linear_range()` must be evenly divisible by `ChunkSize`.
 
-_Effects_: Synchronizes all work-items in the resulting `tangle_group`.
+_Effects:_ Blocks until all work-items in `parent` have reached this
+synchronization point.
 
-_Returns_: A `tangle_group` consisting of the work-items in `group` which are
-part of the same tangle.
+_Synchronization:_ The call in each work-item happens before any work-item
+blocking on the same synchronization point is unblocked.
+Synchronization operations used by an implementation must respect the memory
+scope reported by `ParentGroup::fence_scope`.
 
+_Returns_: A `chunk` consisting of all work-items in `parent` that are in the
+same partition as the calling work-item.
 
-==== `tangle_group` Class
+
+==== `chunk` Class
 
 [source, c++]
 ----
 namespace sycl::ext::oneapi::experimental {
 
-template <typename ParentGroup>
-class tangle_group {
+template <std::size_t ChunkSize, typename ParentGroup>
```

Comment:
We should probably update `linear_id_type` definition here, see the discussion about recorded inconsistency in #13702. Linear id type is smaller than chunk size, meaning that in certain cases we won't be able to enumerate all work-items in a  chunk.

Reply:
I see a few possible fixes here and I'm not sure what is best:

1. `linear_id_type = std::size_t`
Always matches, no problems.

2. `linear_id_type = ParentGroup::linear_id_type`
Sets `uint32_t` for `sub_group` and `size_t` for `group`. The situation where somebody provides a `ChunkSize ` larger than the `ParentGroup` would prevent enumeration of all work-items in the chunk, but that situation is already UB.

3. `linear_id_type = /* smallest unsigned integer with log2(ChunkSize) bits */`
Could use `std::uint8_t` for `ChunkSize` < 256.  Might lead to future ABI issues if we get something like a `std::uint4_t` in future (but that seems unlikely, given the number of bits in a byte).

Reply:
I lean towards (2) here, because it makes `chunk` consistent with its parent group. I.e. if someone had a code for sub-groups, but then introduced some specialization for chunks, types of all IDs they may have in use would still match what APIs return. And this will also be true when we extend chunks to support `group`.

Reply:
Makes sense to me.  I made the change in 0f2a4712ee9fb65c52c6676d3d3a6e3ba55c253c.